dot/state/block.go

```diff
@@ -250,6 +250,37 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetBlocksBySlot gets all blocks that were produced in given slot.
+func (bs *BlockState) GetBlocksBySlot(slotNum uint64) ([]common.Hash, error) {
+	highestFinalisedHash, err := bs.GetHighestFinalisedHash()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get highest finalised hash: %w", err)
+	}
+
+	descendants, err := bs.bt.GetAllDescendants(highestFinalisedHash)
```

Comment:
Does this function work for finalized blocks? Because all finalized blocks are pruned from the memory once finalized and the `blocktree` only contains a reference for blocks in memory (not finalized yet)

Reply:
all descendants will be not finalised